src/Lexer.php

```diff
@@ -44,7 +44,7 @@ class Lexer
     public const STATE_INTERPOLATION = 4;
 
     public const REGEX_NAME = '/[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*/A';
-    public const REGEX_NUMBER = '/[0-9]+(?:\.[0-9]+)?([Ee][\+\-][0-9]+)?/A';
+    public const REGEX_NUMBER = '/[0-9]+(?:_[0-9]+)*(?:\.[0-9]+(?:_[0-9]+)*)?([Ee][\+\-][0-9]+)?/A';
```

Comment:
Maybe we can do the same as in https://github.com/symfony/symfony/blob/7.3/src/Symfony/Component/ExpressionLanguage/Lexer.php#L42-L43 where the exponent can also accept `_`.

Reply:
https://github.com/symfony/symfony/pull/44073